File: app/src/main/java/dev/ragnarok/fenrir/fragment/search/nextfrom/StringNextFrom.java
package dev.ragnarok.fenrir.fragment.search.nextfrom;

import android.os.Parcel;

public class StringNextFrom extends AbsNextFrom {

    public static final Creator<StringNextFrom> CREATOR = new Creator<StringNextFrom>() {
        @Override
        public StringNextFrom createFromParcel(Parcel in) {
            return new StringNextFrom(in);
        }

        @Override
        public StringNextFrom[] newArray(int size) {
            return new StringNextFrom[size];
        }
    };
    private String nextFrom;

    public StringNextFrom(String nextFrom) {
        this.nextFrom = nextFrom;
    }

    protected StringNextFrom(Parcel in) {
        nextFrom = in.readString();
    }

    public String getNextFrom() {
        return nextFrom;
    }

    public void setNextFrom(String nextFrom) {
        this.nextFrom = nextFrom;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(nextFrom);
    }

    @Override
    public void reset() {
        nextFrom = null;
    }
}

File: app/src/main/java/dev/ragnarok/fenrir/fragment/search/nextfrom/StringNextFrom.kt
package dev.ragnarok.fenrir.fragment.search.nextfrom

import android.os.Parcel
import android.os.Parcelable

class StringNextFrom : AbsNextFrom {
    var nextFrom: String?

    constructor(nextFrom: String?) {
        this.nextFrom = nextFrom
    }

    constructor(`in`: Parcel) {
        nextFrom = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(nextFrom)
    }

    override fun reset() {
        nextFrom = null
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<StringNextFrom> =
            object : Parcelable.Creator<StringNextFrom> {
                override fun createFromParcel(`in`: Parcel): StringNextFrom {
                    return StringNextFrom(`in`)
                }

                override fun newArray(size: Int): Array<StringNextFrom?> {
                    return arrayOfNulls(size)
                }
            }
    }
}